### DISCUSSION TITLE
A problem of converting bunge euler angles to {hkil}<uvtw> using `round2Miller`

### DISCUSSION THREAD

**Discussion by PengWei97** [mtex-toolbox/mtex#1633] [Ask Anything]
Dear MTEX experts,

Recently, I tried to get the corresponding {hkil}<uvtw> by using the Euler angles obtained from the basal pole figure in CP-Ti, but there was a mismatch when redrawing the obtained {hkil} on the basal pole figure. So, I don't know if it's because I'm doing something stupid or something else, expecting the MTEX experts to give me an answer.

Specifically, by importing a CTF file and drawing the (0002) basal pole figure in Fig. 1(a), I get the Euler angle  $\left({135.4}^\circ,{74.8}^\circ,{219.2}^\circ\right)$ of point A, and use ` round2Miller` to get its corresponding $\left(\bar{2}\bar{9}\ 11\ 5\right)<10 46 9>$, and finally redraw to $\left(\bar{2}\bar{9}\ 11\ 5\right)$ the $(0002)$ basal pole figure, but found that point A and point B do not coincide as shown in Fig. 1(b).

![image](https://user-images.githubusercontent.com/56183775/200470813-d7b0a762-4dcc-4733-b549-cd37055d392e.png)

Below is the code about the conversion based on #1112,  
```matlab
%% g(phi1, phi, phi2) - {hkil}<uvtw>
  ori_euler = [135.4, 74.8, 219.2]; 
  ori = orientation.byEuler(ori_euler(1)*degree,ori_euler(2)*degree,ori_euler(3)*degree,CS);
  round2Miller(ori,CS)

%% Based on one crystal plane normal direction, obtain all indices in the (0002) basal pole figure
h = Miller({0,0,0,1},{-2,-9,11,5},CS);

figure(2)
for i = 1:length(h)
  plot(h(i),'symmetrised','backgroundColor','w','grid','upper','labeled'); % ,   ,'labeled'
  hold on
end
hold off
```
Finally, the adopted scripts and source files are also uploaded.  [problem1_euler2miller.zip](https://github.com/mtex-toolbox/mtex/files/9957445/problem1_euler2miller.zip)

Any suggestions or recommendations to fix the problem would be greatly appreciated.

Thank you
Wei

**Comment by ralfHielscher** [comment]
Hi Peng,

the command tooltip for the command ```plotPDF(ori)``` does not work, if the option ```'contourf'``` is used. A better way to do what you want is
```matlab
% compute an ODF
odf = calcDensity(ori);

% find the dominante orientation in the ODF
[~,ori_s] = max(odf)

% display the pole figure
plotPDF(odf,Miller(0,0,0,2,CS))

% show the dominate orientation
annotate(ori_s)
```

On ```ori_s``` you can now also use ```round2Miller```
Ralf.

**Reply by PengWei97** [reply]
Yeah, thank you so much for giving me such critical advice.

According to your suggestion, I redrawn the basal plane pole figure, although point A and point B are very close, but there is still a certain deviation. So, my question is this deviation acceptable, or is it an error introduced by using the `round2Miller` transformation?
![image](https://user-images.githubusercontent.com/56183775/200509602-3e1e44d8-bd99-4273-99fb-b3e8cbc2c6be.png)

peng

**Comment by ralfHielscher** [comment]
Sure, there is a deviation. This is why the command is called **round**2Miller. The questions is, why you need to use this command.

Ralf.

**Reply by PengWei97** [reply]
Great，my ultimate goal is to accurately obtain the Miller Bravais index according to the Euler angles, so as to analyze the preferred orientation in a series of pole figures.

Peng.

**Reply by PengWei97** [reply]
And I also have a question, why after calculating the odf, the Miller Bravais  index calculated by the Euler angle can be well matched with the original point in the polar figure, but the Euler angle obtained by the odf is not used, why is the deviation so large? 

Peng
